File: packages/ruby/lib/opensms/resources/sandbox.rb
# frozen_string_literal: true

require_relative "base"

module Opensms
  module Resources
    # The +sandbox+ resource: the rendered text of sandbox sends, including
    # OTP codes. Accessed as +client.sandbox+.
    class Sandbox < Base
      # GET /v1/sandbox/messages -> Page<SandboxMessage>.
      def list_messages(params = nil, **kwargs)
        http_page("/v1/sandbox/messages", Models.build(merge(params, kwargs), %i[limit cursor]))
      end
    end
  end
end

File: packages/python/src/opensms/resources/sandbox.py
"""The ``sandbox`` resource."""

from __future__ import annotations

from typing import Optional

from ..models import Page, SandboxMessage
from ._base import Resource


class Sandbox(Resource):
    """Sandbox inspection. Accessed as ``client.sandbox``."""

    def list_messages(
        self, *, limit: Optional[int] = None, cursor: Optional[str] = None
    ) -> Page[SandboxMessage]:
        """Rendered texts of sandbox sends, including OTP codes."""
        return self._page(
            self._t.request("GET", "/v1/sandbox/messages", query={"limit": limit, "cursor": cursor})
        )
